### PR TITLE
[HDInsight] Onboard new SDK 

### PR DESCRIPTION
**Pull request by aim-for-better**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

**Comment by aim-for-better** [comment]
The SDK namespace board review is https://github.com/Azure/azure-sdk-pr/issues/1247

**Comment by aim-for-better** [comment]
Hi @ArcturusZhang  could you please help review this?  Thanks

File: sdk/hdinsight/Azure.ResourceManager.HDInsight.Containers/src/autorest.md
Comment (ArcturusZhang) [inline sdk/hdinsight/Azure.ResourceManager.HDInsight.Containers/src/autorest.md:15]: Could you please disable this option? We usually do not include information like this into a shipped SDK

File: sdk/hdinsight/Azure.ResourceManager.HDInsight.Containers/src/autorest.md
Comment (ArcturusZhang) [inline sdk/hdinsight/Azure.ResourceManager.HDInsight.Containers/src/autorest.md:9]: To avoid any possible issues, I suggest we use `HDInsightContainers` as library name to disambiguate from `Azure.ResourceManager.HDInsight`

File: sdk/hdinsight/Azure.ResourceManager.HDInsight.Containers/src/autorest.md
Comment (ArcturusZhang) [inline sdk/hdinsight/Azure.ResourceManager.HDInsight.Containers/src/autorest.md:8]: Could we change this to use the readme file as input instead?
https://github.com/Azure/azure-rest-api-specs/blob/f09aacf4c6b63be416212cb182f6b31e8bc6d545/specification/hdinsight/resource-manager/Microsoft.HDInsight/HDInsightOnAks/readme.md
This will enable the automatically update in `azure-sdk-for-net` in the swagger repo if we have a new PR there and preview your changes in dotnet SDK.

**Review by ArcturusZhang — COMMENTED** [review]
A few comments.
I did not see the `api` file, please run the following commands to make sure everything in your PR is up-to-date:
- `dotnet build /t:GenerateCode`
- `dotnet build /t:GenerateTests`
- `<repoRoot>/eng/Export-API.ps1 hdinsight`

File: sdk/hdinsight/Azure.ResourceManager.HDInsight.Containers/src/autorest.md
Comment (aim-for-better) [reply sdk/hdinsight/Azure.ResourceManager.HDInsight.Containers/src/autorest.md:15]: What is the impact of this line?

File: sdk/hdinsight/Azure.ResourceManager.HDInsight.Containers/src/autorest.md
Comment (aim-for-better) [reply sdk/hdinsight/Azure.ResourceManager.HDInsight.Containers/src/autorest.md:8]: Hi @ArcturusZhang , we can not use the readme.md under that commit because the readme.md file need to be updated otherwise the library name or the management client name is not correct. 
The https://github.com/Azure/azure-rest-api-specs/blob/f09aacf4c6b63be416212cb182f6b31e8bc6d545/specification/hdinsight/resource-manager/Microsoft.HDInsight/HDInsightOnAks/readme.md is not the same wit the approved namespace.

I have new PR to update the readme file, it hasn't been merged please see https://github.com/Azure/azure-rest-api-specs/pull/25321

Comment (aim-for-better) [reply sdk/hdinsight/Azure.ResourceManager.HDInsight.Containers/src/autorest.md:8]: I think I can update it once my update readme file PR is merged.

**Comment by aim-for-better** [comment]
> A few comments. I did not see the `api` file, please run the following commands to make sure everything in your PR is up-to-date:
> 
> * `dotnet build /t:GenerateCode`
> * `dotnet build /t:GenerateTests`
> * `<repoRoot>/eng/Export-API.ps1 hdinsight`

Hi @ArcturusZhang , thanks for your review. Yes, I didn't run export-api.ps1, now I have added the api folder

File: sdk/hdinsight/Azure.ResourceManager.HDInsight.Containers/src/autorest.md
Comment (aim-for-better) [reply sdk/hdinsight/Azure.ResourceManager.HDInsight.Containers/src/autorest.md]: removed this config

File: sdk/hdinsight/Azure.ResourceManager.HDInsight.Containers/src/autorest.md
Comment (aim-for-better) [reply sdk/hdinsight/Azure.ResourceManager.HDInsight.Containers/src/autorest.md]: updated it to HDInsightContainers

File: sdk/hdinsight/Azure.ResourceManager.HDInsight.Containers/src/autorest.md
Comment (aim-for-better) [reply sdk/hdinsight/Azure.ResourceManager.HDInsight.Containers/src/autorest.md]: Good news is that the update readme file is merged today. I though it will take several days. Updated.

**Comment by aim-for-better** [comment]
Hi @ArcturusZhang  could you please help review it again? Thanks

File: sdk/hdinsight/Azure.ResourceManager.HDInsight.Containers/src/autorest.md
Comment (ArcturusZhang) [inline sdk/hdinsight/Azure.ResourceManager.HDInsight.Containers/src/autorest.md:13]: Could you please also include the content of `autorest.md` in the template starting from line 19 to the end of the template file?
These contents are required to generate the SDKs.
Here is the template: https://github.com/Azure/azure-sdk-for-net/blob/2c1d884cc72b7af6ef1606396802a8059a687dee/eng/templates/Azure.ResourceManager.Template/src/autorest.md?plain=1#L19

Comment (aim-for-better) [reply sdk/hdinsight/Azure.ResourceManager.HDInsight.Containers/src/autorest.md]: Hi @ArcturusZhang , 
I respect your comment and did some test work after received this comment.

However, I found one error if we apply the 
format-by-name-rules:  *Uri': 'Uri'.  The impact of applying this is the SDK can't work due to the Uri end with "/" , but our RP doesn't support.

For this time, I hope that we can skip this. Thanks.

**Comment by aim-for-better** [comment]
Hi @ArthurMa1978 , I am sorry that you can not merge this commit https://github.com/Azure/azure-sdk-for-net/pull/38256/commits/758855f7bb8966417e35b7c311a1ef7a08c6d6e2

The reason is that our swagger spec requires it is string with format validation, if we change the string to Sytem.Uri, in http request it will end with "/", this request will fail, the SDK won't work.

File: sdk/hdinsight/Azure.ResourceManager.HDInsight.Containers/src/autorest.md
Comment (aim-for-better) [reply sdk/hdinsight/Azure.ResourceManager.HDInsight.Containers/src/autorest.md:13]: Hi @ArcturusZhang , I applied the rules excepts for the Uri part, thanks

**Comment by aim-for-better** [comment]
Hi @ArthurMa1978 could you please help continue to review this PR?
I have addressed the comment with the suggestion from @ArcturusZhang .  Thanks

**Review by ArthurMa1978 — APPROVED** [review]

**Pull request merged**
